src/utils/strings.js

```diff
@@ -40,16 +40,83 @@ export function escapeHtml(str) {
   return str.replace(HTML_UNSAFE_RE, sanitize);
 }
 
-// Add the correct grammatical number (singular or plural)
-// to a noun depending on the number of items of an array
-export function pluralize(str, array) {
-  let plural;
-  if (str.slice(-1) === 'y') {
-    plural = `${str.slice(0, -1)}ies`;
-  } else {
-    plural = `${str}s`;
+export const PluralCategory = {
+  zero: 'zero',
+  one: 'one',
+  two: 'two',
+  few: 'few',
+  many: 'many',
+  other: 'other',
+};
+
+export const PluralRuleType = {
+  cardinal: 'cardinal',
+  ordinal: 'ordinal',
+};
+
+// Returns the "pluralized" version of some provided text based on an integer
+// count and the user's resolved locale.
+//
+// Choices _must_ be provided for the en locale since that will be used as a
+// fallback in the scenario where no choices are provided or available for the
+// resolved locale of a given user.
+//
+// @param {Object} choices An object keyed by locales. Each value should itself
+//   be an object with keys for each plural category supported by the locale
+//   with the value representing the string to return.
+//
+// Examples:
+//
+//   const choices = {
+//     en: {
+//       one: 'day',
+//       other: 'days',
+//     },
+//     sl: {
+//       one: 'ura',
+//       two: 'uri',
+//       few: 'ure',
+//       other: 'ur',
+//     },
+//   };
+//
+//   // en
+//   pluralize(choices, 2); // 'days'
+//   pluralize({ en: { one: 'Technology, other: 'Technologies' } }, 1) // 'Technology'
+//   pluralize({}, 1) // throws error
+//
+//   // sl
+//   pluralize(choices, 2); // 'uri'
+//   pluralize({ sl }, 2); // throws error
+//   pluralize({ en }, 2); // 'days'
+export function pluralize(choices, count) {
+  const { cardinal } = PluralRuleType;
+  const { one, other } = PluralCategory;
+
+  const defaultLocale = 'en';
+  const defaultCategory = count === 1 ? one : other;
+
+  // at minimum, there should at least be a "one" and "other" choice for the
+  // "en" locale for use as fallback text in the case that a choice for the
+  // user's resolved locale category is not provided/available
+  if (!choices[defaultLocale] || !choices[defaultLocale][defaultCategory]) {
```

Comment:
Should we allow for just specifying `other`, so we dont duplicate `one` and `other`, in cases like the Extends?

Reply:
I'm not sure to be honest. My initial thought is no, because the `pluralize` function isn't smart enough to know that we want to use the same word for both cases in that specific scenario. I think it should have the expectation that each language provided in the `choices` argument should have text to use for every plural category that the specific language may need. We could hardcode in special logic for this scenario in English maybe, but I think I would prefer to avoid that—maybe we could have a helper function for generating the duplicate entries if this ends up being a common enough problem?

It felt a little weird to even have this hardcoded logic checking for all the English options, but I think that makes sense for our app since we'll want some default language to show in every place where this function is used, even if we don't have a particular translation for some other language (in the future).

Reply:
Also, thinking about it a bit more—it's more of a weird thing we're doing in the code for this situation since "Extends" isn't really a noun that we want to potentially pluralize like we do for Framework/Technology, so potentially it would make more sense to provide the ability to pass a static title that doesn't get pluralized for that type of situation. I think it's fine as-is for now though.